versioned_docs/version-v1.19.x/extend/extend-apiml/implement-new-saf-provider.md

```diff
@@ -0,0 +1,89 @@
+# Implement new SAF IDT provider
+
+API Gateway offers the capability to implement your own SAF Identity Token (IDT) provider, by implementing 
+an existing interface.
+
+- [How to create the SAF IDT provider](#how-to-create-the-saf-idt-provider)
+- [How to integrate your extension to API ML](#how-to-integrate-your-extension-to-api-ml)
+- [How to use the SAF IDT provider](#how-to-use-the-saf-idt-provider)
+
+## How to create the SAF IDT provider
+
+If you want to create your own implementation of the SAF IDT provider, you can implement the existing `org.zowe.apiml.gateway.security.service.saf.SafIdtProvider` interface. 
+
+```java
+public interface SafIdtProvider {
+    /**
+     * If the current user has the proper rights generate the SAF token on its behalf and return it back.
+     *
+     * @return Either empty answer meaning the user is either unauthenticated or doesn't have the proper rights.
+     */
+    Optional<String> generate(String username);
+
+    /**
+     * Verify that the provided saf token is valid.
+     *
+     * @param safToken Token to validate.
+     * @return true if the token is valid, false if it is invalid
+     */
+    boolean verify(String safToken);
+}
+```
+
+The `SafIdtProvider` interface contains the `generate` and `verify` methods, that can be overridden by your SAF IDT implementation
```

Comment:
Check this for accuracy: The `SafIdtProvider` interface contains the `generate` and `verify` methods. These methods can be overridden by your SAF IDT implementation to generate the SAF token on behalf of the specified user. This also verifies that the provided SAF token is valid.

Reply:
Also, what is verifying that the SAF token is valid?

Reply:
The SAF IDT implementation verify that the SAF token is valid. I think that your reformulation is correct